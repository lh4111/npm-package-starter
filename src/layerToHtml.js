export default function layerToHtml (layer) {
  const { _class, name, frame: { x, y, width, height } } = layer
  return `<div id="${_class}_${name}" style="position: absolute; top: ${y}px; left: ${x}px; width: ${width}px; height: ${height}px"></div>`
}
