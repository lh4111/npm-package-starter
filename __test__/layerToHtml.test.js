import layerToHtml from '../src/layerToHtml'

describe('layer to html', () => {
  it('render layer to html', () => {
    const layer = {
      _class: 'Shape',
      name: 'testShape',
      frame: {
        x: 100,
        y: 100,
        width: 200,
        height: 200
      }
    }
    expect(layerToHtml(layer)).toMatchSnapshot()
  });
});
