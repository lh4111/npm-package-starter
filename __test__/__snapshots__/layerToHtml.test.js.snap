// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layer to html render layer to html 1`] = `"<div id=\\"Shape_testShape\\" style=\\"position: absolute; top: 100px; left: 100px; width: 200px; height: 200px\\"></div>"`;
